### PR TITLE
Update edit command syntax

### LINKED ISSUE #11: As a standard user, I want to edit existing entries
... so that I can update entries without having to recreate them.

### LINKED ISSUE #149: isSameSupplier/Warehouse method in supplier/warehouse class
I noticed that for supplier, it is a comparison by name and either matching phone/email address. However, for warehouse, it is only a comparison by name. While updating the edit command to reflect error when user tries to edit an entry with the same name as another existing supplier, I noticed this issue and have created new method to compare by Name only for supplier. In this case, should we standardise how we are comparing under isSameSupplier/Warehouse method?

### PR DESCRIPTION
**Pull request by tohyuting**
Updated edit command syntax to fit the new standardised style we agreed upon in the previous meeting. Changes for the tests have been made as well.

### CODE REVIEW

**Review by jeffreytjs — COMMENTED** [review]
LGTM, good coverage with the tests cases. Nice one maintaining backward compatibility too, we used the same names even by adding 2!

**Comment by tohyuting** [comment]
Thanks for the review! I will be pushing up another commit to fix some logical issues and to incorporate Type class into my code :)

**Comment by tohyuting** [comment]
Will be adding testcases for isSameSupplierByName method if we intend to keep this method in the end instead of changing isSameSupplier.

File: src/main/java/seedu/clinic/logic/parser/EditCommandParser.java [hunk 1]
```diff
@@ -45,51 +46,56 @@
...
+                ArgumentTokenizer.tokenize(args, PREFIX_TYPE, PREFIX_INDEX, PREFIX_NAME,
                         PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_REMARK);
 
         logger.log(Level.INFO, "Successfully tokenized user input.");
 
-        if (argMultimap.getValue(PREFIX_SUPPLIER_INDEX).isPresent()
-                && argMultimap.getValue(PREFIX_WAREHOUSE_INDEX).isPresent()) {
-            throw new ParseException(String.format(MESSAGE_INPUT_BOTH_SUPPLIER_WAREHOUSE_PREFIX,
+        if (!argMultimap.getValue(PREFIX_TYPE).isPresent()
+                && !argMultimap.getValue(PREFIX_INDEX).isPresent()) {
```
Comment (Criss-Wang) [inline src/main/java/seedu/clinic/logic/parser/EditCommandParser.java:55]: I think the original way is to define a new function called arePrefixsPresent (maybe you can check the deleteCommandParser and addCommandParser). Not sure if this is the new form intended

File: src/main/java/seedu/clinic/logic/parser/EditCommandParser.java [hunk 2]
```diff
@@ -203,4 +181,21 @@ private EditDescriptor parseGeneralDetails(EditDescriptor editDescriptor, Argume
...
+            return new ParseException(String.format(MESSAGE_INVALID_USAGE, EditCommand.MESSAGE_USAGE));
+        }
+        if (prefix.equals(PREFIX_TYPE)) {
+            return new ParseException(String.format(MESSAGE_INVALID_TYPE_EDIT, EditCommand.MESSAGE_USAGE));
+        } else if (prefix.equals(PREFIX_INDEX)) {
+            return new ParseException(MESSAGE_INVALID_INDEX + "\n" + EditCommand.MESSAGE_USAGE);
+        } else {
+            assert prefix.equals(PREFIX_PHONE) : "The prefix here should be of Phone type!";
+            return new ParseException(Phone.MESSAGE_CONSTRAINTS + "\n" + EditCommand.MESSAGE_USAGE);
+        }
```
Comment (Criss-Wang) [inline src/main/java/seedu/clinic/logic/parser/EditCommandParser.java:199]: May I know the intention of this method? I though the checking of invalid argument is done during the class construction itself, for example:
 ```
public Address(String address) {
        requireNonNull(address);
        checkArgument(isValidAddress(address), MESSAGE_CONSTRAINTS);
        value = address;
    }
```

**Review by Criss-Wang — COMMENTED** [review]
I'm not quite sure about some parts in the code, maybe adding some comment above the method will help.

File: src/main/java/seedu/clinic/logic/parser/EditCommandParser.java (hunk 1 above)
Comment (tohyuting) [reply src/main/java/seedu/clinic/logic/parser/EditCommandParser.java:55]: Just went to take a look at how it is defined in deleteCommandParser and addCommandParser, and yes I think arePrefixsPresent does these checks! I can change it to this function, However, I realised that would also mean I will not be able to throw specific ParseException reminding user which prefixes they are missing (either one, or all). Which do you think would be better (in terms of code quality and user experience?) :)

File: src/main/java/seedu/clinic/logic/parser/EditCommandParser.java (hunk 2 above)
Comment (tohyuting) [reply src/main/java/seedu/clinic/logic/parser/EditCommandParser.java:199]: I can remove this if we want to just show a more generic error message instead. For instance, if someone entered p/99999999 z/testing, using the check in class construction will show an error message that the phone number is not valid. However, this is not the case as the number attached to prefix p is valid, except an incorrect prefix was added at the end, resulting in the check for isValidPhone to fail. Same for checks where p/99999999 testing when user enters random string value behind a prefix (or will this case be treated as wrong input?)

**Review by zhengweii — COMMENTED** [review]
LGTM

**Review by zhengweii — APPROVED** [review]

**Pull request merged**